File: csharp/Behavioral/19-State/State/PausedState.cs
namespace State;

public class PausedState : BaseJobState
{
    public override string Name => "Paused";

    public override void Resume(JobCard job)
    {
        job.PauseReason = null;
        job.TransitionTo(new InProgressState());
        Console.WriteLine("  ▶️  Resumed");
    }
}

File: typescript/Behavioral/19-State/State/PausedState.ts
import { BaseJobState } from "./BaseJobState";
import { InProgressState } from "./InProgressState";
import type { JobCard } from "../Context/JobCard";

export class PausedState extends BaseJobState {
  readonly name = "Paused";

  override resume(job: JobCard): void {
    job.pauseReason = undefined;
    job.transitionTo(new InProgressState());
    console.log("  ▶️  Resumed");
  }
}
